backend/src/akvo/lumen/admin/add_tenant.clj

```diff
@@ -35,8 +69,115 @@
+        pair-resp (client/post
+                   (format "%s/groups/%s/role-mappings/realm" api-root new-group-id)
+                   (assoc request-draft
+                          :body (json/encode [{"id" role-id
+                                               "name" role
+                                               "scopeParamRequired" false
+                                               "composite" false
+                                               "clientRole" false
+                                               "containerId" "Akvo"}])))]
+    new-group-id))
+
+(defn create-new-user
+  [request-draft api-root email]
+  (let [tmp-password (random-url-safe-string 6)
+        user-id (-> (client/post (format "%s/users" api-root)
+                                 (assoc request-draft
+                                        :body (json/encode
+                                               {"username" email
+                                                "email" email
+                                                "emailVerified" false
+                                                "enabled" true})))
+                    (get-in [:headers "Location"])
+                    (s/split #"/")
+                    last)]
+    (client/put (format "%s/users/%s/reset-password" api-root user-id)
+                (assoc request-draft
+                       :body (json/encode {"temporary" true
+                                           "type" "password"
+                                           "value" tmp-password})))
+    {:email email
+     :user-id user-id
+     :tmp-password tmp-password}))
+
+(defn user-representation
+  [request-draft api-root email]
+  (if-let [{:strs [id] :as user} (keycloak/fetch-user-by-email request-draft
+                                                               api-root email)]
+    {:email email
+     :user-id id}
+    (create-new-user request-draft api-root email)))
+
+(defn setup-keycloak
+  [label email]
+  (let [{api-root :api-root :as kc} (util/create-keycloak)
+        request-draft (keycloak/request-draft kc)
+        lumen-group-id (root-group-id request-draft api-root)
+        tenant-id (create-group request-draft api-root lumen-group-id
+                                (format "akvo:lumen:%s" label) label)
+        tenant-admin-id (create-group request-draft api-root tenant-id
+                                      (format "akvo:lumen:%s:admin" label)
+                                      "admin")
+        {:keys [user-id email tmp-password] :as user-rep}
+        (user-representation request-draft api-root email)]
+    (keycloak/add-user-to-group request-draft api-root user-id tenant-admin-id)
+    user-rep))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Main
+;;;
+
+(defn -main [label title email]
+  (try
+    (setup-database (conform-label label) title)
+    (let [user-creds (setup-keycloak label email)]
+      (println "User creds:")
```

Comment:
Thinking about it maybe this should have been automated by sending an email in the script.